File: 1962-remove-stones-to-minimize-the-total/1962-remove-stones-to-minimize-the-total.cpp
class Solution {
public:
  int minStoneSum(vector<int>& piles, int k) {
    priority_queue<int> pq;
    for (auto &pile : piles) pq.push(pile);
    while (k--) {
      int p = pq.top();
      pq.pop();
      pq.push(p-p/2);
    }
    int s = 0;
    while (!pq.empty()) {
      s += pq.top();
      pq.pop();
    }
    return s;
  }
};

File: 1962-remove-stones-to-minimize-the-total/1962-remove-stones-to-minimize-the-total.py
class Solution:
  def minStoneSum(self, piles: List[int], k: int) -> int:
    h = []
    for pile in piles:
      heappush(h, -pile)
    for _ in range(k):
      heappush(h, heappop(h)//2)
    return -sum(h)